wasm/jit/jit_amd64_test.go

```diff
@@ -3877,6 +3877,215 @@ func TestAmd64Compiler_compileRem(t *testing.T) {
+						} else {
+							expFloat64 = math.Abs(math.Float64frombits(v))
+						}
+					case *wazeroir.OperationNeg:
+						compileOperationFunc = func() {
+							err := compiler.compileNeg(o)
+							require.NoError(t, err)
+						}
+						is32Bit = o.Type == wazeroir.Float32
+						if is32Bit {
+							expFloat32 = -math.Float32frombits(uint32(v))
+						} else {
+							expFloat64 = -math.Float64frombits(v)
+						}
+					case *wazeroir.OperationCeil:
+						compileOperationFunc = func() {
+							err := compiler.compileCeil(o)
+							require.NoError(t, err)
+						}
+						is32Bit = o.Type == wazeroir.Float32
+						if is32Bit {
+							expFloat32 = float32(math.Ceil(float64(math.Float32frombits(uint32(v)))))
+						} else {
+							expFloat64 = math.Ceil(math.Float64frombits(v))
+						}
+					case *wazeroir.OperationFloor:
+						compileOperationFunc = func() {
+							err := compiler.compileFloor(o)
+							require.NoError(t, err)
+						}
+						is32Bit = o.Type == wazeroir.Float32
+						if is32Bit {
+							expFloat32 = float32(math.Floor(float64(math.Float32frombits(uint32(v)))))
+						} else {
+							expFloat64 = math.Floor(math.Float64frombits(v))
+						}
+					case *wazeroir.OperationTrunc:
+						compileOperationFunc = func() {
+							err := compiler.compileTrunc(o)
+							require.NoError(t, err)
+						}
+						is32Bit = o.Type == wazeroir.Float32
+						if is32Bit {
+							expFloat32 = float32(math.Trunc(float64(math.Float32frombits(uint32(v)))))
+						} else {
+							expFloat64 = math.Trunc(math.Float64frombits(v))
+						}
+					case *wazeroir.OperationSqrt:
+						compileOperationFunc = func() {
+							err := compiler.compileSqrt(o)
+							require.NoError(t, err)
+						}
+						is32Bit = o.Type == wazeroir.Float32
+						if is32Bit {
+							expFloat32 = float32(math.Sqrt(float64(math.Float32frombits(uint32(v)))))
+						} else {
+							expFloat64 = math.Sqrt(math.Float64frombits(v))
+						}
+					case *wazeroir.OperationNearest:
+						compileOperationFunc = func() {
+							err := compiler.compileNearest(o)
+							require.NoError(t, err)
+						}
+						is32Bit = o.Type == wazeroir.Float32
+						if is32Bit {
+							// Implementation from https://github.com/wasmerio/wasmer/blob/703bb4ee2ffb17b2929a194fc045a7e351b696e2/lib/vm/src/libcalls.rs#L77.
+							f := math.Float32frombits(uint32(v))
+							f64 := float64(f)
+							if f != -0 && f != 0 {
+								u := float32(math.Ceil(f64))
+								d := float32(math.Floor(f64))
+								um := math.Abs(float64(f - u))
+								dm := math.Abs(float64(f - d))
+								h := u / 2.0
+								if um < dm || (um == dm && float32(math.Floor(float64(h))) == h) {
+									f = u
+								} else {
+									f = d
+								}
+							}
+							expFloat32 = f
+						} else {
+							// Implementation from https://github.com/wasmerio/wasmer/blob/703bb4ee2ffb17b2929a194fc045a7e351b696e2/lib/vm/src/libcalls.rs#L120-L144.
```

Comment:
can you just re-implement these two? especially the variable names aren't great and I think we can do better

Reply:
re-implemented! https://github.com/tetratelabs/wazero/pull/119/commits/d6fc64a40917c4bb0b4277876a4c5b69afee680d